redfish/jobservice.go

```diff
@@ -0,0 +1,129 @@
+	MaxSteps int
+	// ServiceEnabled This property shall indicate whether the Schedule property within the job supports scheduling of jobs.
+	ServiceEnabled bool
+}
+
+// JobService is used to represent the job service offered by the redfish API
+type JobService struct {
+	common.Entity
+
+	// ODataContext is the odata context.
+	ODataContext string `json:"@odata.context"`
+	// ODataType is the odata type.
+	ODataType string `json:"@odata.type"`
+	// Description provides a description of this resource.
+	Description string
+	// DateTime system time.
+	DateTime time.Time
+	// jobs points towards the jobs collection
+	jobs string
+	// jobs points towards the jobs collection
+	// TODO: Add log collection fetching
+	log string
+
+	// ServiceCapabilities This type shall contain properties that describe the capabilities or supported features of this implementation of a job service.
+	ServiceCapabilities ServiceCapabilities
+
+	// ServiceEnabled indicates whether this service isenabled.
+	ServiceEnabled bool
+	// Status describes the status and health of a resource and its children.
+	Status common.Status
+}
+
+// GetJob will get a Job instance from the service.
+func GetJob(c common.Client, uri string) (*Job, error) {
+	var job Job
+	return &job, job.Get(c, uri, &job)
+}
+
+// Jobs will get collection of Job from the service.
+func (jobService *JobService) Jobs() ([]*Job, error) {
+	return jobService.listReferencedJobs(jobService.GetClient(), jobService.jobs)
+}
+
+// ListReferencedJobs gets the collection of Job from a provided reference.
+func (jobService *JobService) listReferencedJobs(c common.Client, link string) ([]*Job, error) {
+	var result []*Job
+	if link == "" {
+		return result, nil
+	}
+
+	type GetResult struct {
+		Item  *Job
+		Link  string
+		Error error
+	}
+
+	ch := make(chan GetResult)
+	collectionError := common.NewCollectionError()
+	get := func(link string) {
+		job, err := GetJob(c, link)
+		ch <- GetResult{Item: job, Link: link, Error: err}
+	}
+
+	go func() {
+		err := common.CollectList(get, c, link)
+		if err != nil {
+			collectionError.Failures[link] = err
+		}
+		close(ch)
+	}()
+
+	for r := range ch {
+		if r.Error != nil {
+			collectionError.Failures[r.Link] = r.Error
+		} else {
+			result = append(result, r.Item)
+		}
+	}
+
+	if collectionError.Empty() {
+		return result, nil
+	}
+
+	return result, collectionError
+}
+
+// UnmarshalJSON unmarshals a JobService object from the raw JSON.
+func (jobService *JobService) UnmarshalJSON(b []byte) error {
+	type temp JobService
+	var t struct {
+		temp
+		Jobs common.Link
+	}
+
+	err := json.Unmarshal(b, &t)
+	if err != nil {
+		return err
+	}
+
+	// Extract the links to other entities for later
+	*jobService = JobService(t.temp)
+	jobService.jobs = t.Jobs.String()
+
+	return nil
+}
+
+// DeleteJob deletes a specific job using the job service.
+func (jobService *JobService) DeleteJob(uri string) error {
```

Comment:
This should probably be changed to a method on the `Job` object itself so someone can just call `job.Delete()`. There are a few cases where it can't be avoided, but in general we should try to avoid cases where the consumer needs to handle URI's directly.